### ISSUE TITLE
Bug: "stack --resolver=nightly-2015-10-28 install hlint" fails

### ISSUE THREAD

**Issue by rvion** [commercialhaskell/stack#1250]
`stack --resolver=nightly-2015-10-28 install hlint` is failing for me

#### Stack version

```shell
➜  ~  stack --version
Version 0.1.7.0, Git revision 8ea675e97bdff22749f3e5695ae110fae4fb114a x86_64
```

#### uname -a

```shell
➜  ~  uname -a
Darwin xxxxx.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64
```

#### stack --resolver=nightly-2015-10-28 install hlint

```shell
➜  ~  stack --resolver=nightly-2015-10-28 install hlint
Run from outside a project, using implicit global project config
Using resolver: nightly-2015-10-28 specified on command line
After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
```

#### stack --verbose --resolver=nightly-2015-10-28 install hlint

```shell
➜  ~  stack --verbose --resolver=nightly-2015-10-28 install hlint
Version 0.1.7.0, Git revision 8ea675e97bdff22749f3e5695ae110fae4fb114a x86_64
2015-10-28 11:18:33.795509: [debug] Checking for project config at: /Users/RemiVion/stack.yaml @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:554:9)
2015-10-28 11:18:33.796736: [debug] Checking for project config at: /Users/stack.yaml @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:554:9)
2015-10-28 11:18:33.796909: [debug] Checking for project config at: /stack.yaml @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:554:9)
2015-10-28 11:18:33.797148: [debug] No project config file found, using defaults. @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:581:13)
2015-10-28 11:18:33.800514: [info] Run from outside a project, using implicit global project config @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:307:13)
2015-10-28 11:18:33.808803: [info] Using resolver: nightly-2015-10-28 specified on command line @(stack_FdND4vWuZuq6UFAvcXg9mw:Stack.Config src/Stack/Config.hs:331:32)
2015-10-28 11:18:33.809063: [debug] Trying to decode /Users/RemiVion/.stack/build-plan-cache/x86_64-osx/nightly-2015-10-28.cache @(stack_FdND4vWuZuq6UFAvcXg9mw:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:53:5)
2015-10-28 11:18:33.84322: [debug] Success decoding /Users/RemiVion/.stack/build-plan-cache/x86_64-osx/nightly-2015-10-28.cache @(stack_FdND4vWuZuq6UFAvcXg9mw:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:62:13)
2015-10-28 11:18:33.846886: [debug] Run process: ghc --info @(stack_FdND4vWuZuq6UFAvcXg9mw:System.Process.Read src/System/Process/Read.hs:272:3)
2015-10-28 11:18:34.394681: [debug] Run process: ghc --numeric-version @(stack_FdND4vWuZuq6UFAvcXg9mw:System.Process.Read src/System/Process/Read.hs:272:3)
2015-10-28 11:18:34.437228: [debug] Run process: ghc-pkg --no-user-package-db field --simple-output Cabal version @(stack_FdND4vWuZuq6UFAvcXg9mw:System.Process.Read src/System/Process/Read.hs:272:3)
2015-10-28 11:18:34.43743: [debug] Run process: ghc-pkg --no-user-package-db field --simple-output Cabal version @(stack_FdND4vWuZuq6UFAvcXg9mw:System.Process.Read src/System/Process/Read.hs:272:3)
After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
```

Is there anything else I can provide to help troobleshooting this ?

**Comment by snoyberg** [comment]
Should be fixed with 5cc457e13fcedf326a7bb48a87ed4637ab873980, can you try upgrading to latest HEAD?

**Comment by rvion** [comment]
It's working perfectly with master, thanks ! :+1:
